parallel/src/main.rs

```diff
@@ -0,0 +1,45 @@
+use std::sync::mpsc;
+use std::thread;
+use rand::Rng;
+
+fn main() {
+    let (tx, rx) = mpsc::channel();
+    const N: usize = 100; 
+    let mut xs: [i32; N] = [0; N];
+
+    for i in 0..10 {
+        xs[i] = rand::thread_rng().gen_range(1, 101);
+    }
+
+    let mut s: i32 = 0;
+
+    let tx1 = mpsc::Sender::clone(&tx);
+    let first = thread::spawn(move || {
+        let mut sum: i32 = 0;
+        for i in 0..(N/2){
```

Comment:
actually you are copy pasting logic here, you can have a standlone function that accepts some parameters like

 * a reference to data
 * a range to sum over
 * a sender for the mpsc channel